### PR TITLE
Add get all editions endpoint

### PR DESCRIPTION
**Pull request by jondewijones**
### What

Add get all editions endpoint

### Who can review

Anyone

### CODE REVIEW

File: dataset/get-editions.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	err := checkAccessTokenAndCollectionHeaders(userAccessToken, collectionID)
+	if err != nil {
+		log.Event(ctx, err.Error(), log.ERROR)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	log.Event(ctx, "calling get editions", log.Data(logInfo))
```
Comment (CarlHembrough) [inline dataset/get-editions.go]: Missing log level. The same is the case for all logs in this service that start with "calling get"

File: dataset/get-editions.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	})
+}
+
+func getEditions(w http.ResponseWriter, req *http.Request, dc DatasetClient, userAccessToken, collectionID, lang string) {
+	ctx := req.Context()
+
+	vars := mux.Vars(req)
+	datasetID := vars["datasetID"]
+
+	logInfo := map[string]interface{}{
```
Comment (CarlHembrough) [inline dataset/get-editions.go]: Consider adding the collection ID to the log info - it may help debugging?

File: dataset/get-editions.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	b, err := json.Marshal(mapped)
+	if err != nil {
+		log.Event(ctx, "error marshalling response to json", log.ERROR, log.Error(err), log.Data(logInfo))
+		http.Error(w, "error marshalling response to json", http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(b)
+
+	log.Event(ctx, "get versions: request successful", log.INFO, log.Data(logInfo))
```
Comment (CarlHembrough) [inline dataset/get-editions.go]:
```suggestion
	log.Event(ctx, "get editions: request successful", log.INFO, log.Data(logInfo))
```

File: dataset/get-editions.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			continue
+		}
+		latestVersionInEdition[edition.Edition] = version.ReleaseDate
+	}
+
+	mapped := mapper.AllEditions(ctx, dataset, editions, latestVersionInEdition)
+
+	b, err := json.Marshal(mapped)
+	if err != nil {
+		log.Event(ctx, "error marshalling response to json", log.ERROR, log.Error(err), log.Data(logInfo))
```
Comment (CarlHembrough) [inline dataset/get-editions.go]: Consider adding the context for this log
```suggestion
		log.Event(ctx, "error marshalling editions response to json", log.ERROR, log.Error(err), log.Data(logInfo))
```

**Review by CarlHembrough — CHANGES_REQUESTED** [review]
Looks good overall, just some minor log tweaks 👍

File: dataset/get-editions.go (hunk 1 above)
Comment (jondewijones) [reply dataset/get-editions.go]: 👍

File: dataset/get-editions.go (hunk 2 above)
Comment (jondewijones) [reply dataset/get-editions.go]: Added and moved to below where we check if collectionID is set

File: dataset/get-editions.go (hunk 3 above)
Comment (jondewijones) [reply dataset/get-editions.go]: 👍

File: dataset/get-editions.go (hunk 4 above)
Comment (jondewijones) [reply dataset/get-editions.go]: 👍

**Review by CarlHembrough — APPROVED** [review]
LGTM 👍

**Pull request merged**